modules/simple_forms_api/docs/rfcs/0002-async-s3-pdf-upload.md

```diff
@@ -0,0 +1,120 @@
+# RFC: Transitioning PDF Uploads to Asynchronous Sidekiq Jobs
+
+## 1. Title
+
+Transitioning PDF Uploads to Asynchronous Sidekiq Jobs
+
+## 2. Summary
+
+This document proposes transitioning the PDF upload process from a synchronous workflow to an asynchronous Sidekiq job within the VA.gov platform. By moving to asynchronous processing, we aim to improve user experience on the form submission Confirmation page by reducing wait times, while also enhancing error handling and resilience during S3 outages. The RFC outlines the technical requirements, necessary frontend changes, error-handling strategies, security, and performance considerations for implementing this change.
+
+## 3. Background
+
+Currently, PDFs are generated and uploaded synchronously to the `vff-simple-forms` S3 bucket when users submit forms. This process causes delays on the Confirmation page, as users must wait for the upload to complete before accessing their submission PDF. Additionally, any failures during the upload result in immediate user-facing errors, with limited retry or resilience mechanisms in place. This RFC proposes handling the upload asynchronously with Sidekiq, which would enable faster form submissions, provide more robust error handling, and improve the user experience.
+
+## 4. Proposal
+
+### 4.1 Asynchronous Upload with Sidekiq
+
+- **Backend Modifications**:
+  - Modify the `UploadsController#submit` action to trigger a Sidekiq job for PDF uploads, allowing form submissions to complete without waiting for the PDF upload.
+  - The Sidekiq job will handle the PDF upload to S3 and return a presigned URL to the backend, where it will be made available for the frontend to display.
+  - Retry Limit: Set to 3 retries in case of transient errors.
+  - Failure Handling: Log errors and notify users if upload fails, without displaying technical error details.
+
+- **Frontend Modifications**:
```

Comment:
I imagine the backend would provide a url with the initial response 200 of the submit, so frontend can use that url to make the second call to fetch or poll the PDF. Does that align with your thinking?